airflow/models.py

```diff
@@ -826,12 +832,16 @@ def are_dependencies_met(
                 TI.state == State.SUCCESS,
             ).first()
             if not previous_ti:
+                if verbose:
+                    logging.warning("depends_on_past not satisfied")
                 return False
 
             # Applying wait_for_downstream
             previous_ti.task = self.task
             if task.wait_for_downstream and not \
                     previous_ti.are_dependents_done(session):
+                if verbose:
+                    logging.warning("wait_for_downstream not satisfied")
```

Comment:
yes, I get it now, 'wait_for_downstream' could be more precisely called 'wait_for_past_downstream'.